### PR TITLE
Fix warnings and set warning as hard errors

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: lib/shuffle.js [hunk 1]
```diff
@@ -115,8 +115,8 @@ module.exports = function shuffle(array) {
     for (let i = array.length - 1; i > 0; i--) {
         const randIndex = randomRange(0, i);
         const randIndexVal = array[randIndex];
-        array[randIndex] = array[i];
-        array[i] = randIndexVal;
+        array[randIndex] = array[i]; // eslint-disable-line no-param-reassign
```
Comment (michael-zapata) [inline lib/shuffle.js]: This is sad, I'll try to get rid of it later, since we're using the returned array.

Comment (michael-zapata) [inline lib/shuffle.js]: (Also there's a shuffle available in Arsenal that could help as well.)

File: tests/unit/sproxyd.js [hunk 2]
```diff
@@ -16,11 +16,11 @@ const upload = crypto.randomBytes(9000);
 let client;
 let savedKey;
 let server;
-let md = {};
+let md = {}; // eslint-disable-line prefer-const
```
Comment (michael-zapata) [inline tests/unit/sproxyd.js]: You can `const` an object and still be able to edit the fields.

File: tests/unit/sproxyd.js [hunk 3]
```diff
@@ -84,13 +88,13 @@ function handler(req, res) {
     }
 }
 
-client = new Sproxy({ bootstrap: [ '127.0.0.1:9000' ] });
+// eslint-disable-next-line prefer-const
```
Comment (michael-zapata) [inline tests/unit/sproxyd.js]: This is strange. I think we can directly declare it as const as well.

File: package.json [hunk 4]
```diff
@@ -16,7 +17,10 @@
     "werelogs": "scality/werelogs"
   },
   "devDependencies": {
-    "mocha": "^2.3.4"
+    "mocha": "^2.3.4",
```
Comment (michael-zapata) [inline package.json]: Nit: can we keep the list ordered?

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by michael-zapata** [comment]
What about arsenal's shuffle?

**Comment by GuillaumeGomez** [comment]
cc @LaurenSpiegel

**Comment by LaurenSpiegel** [comment]
@GuillaumeGomez please run an end to end

**Comment by GuillaumeGomez** [comment]
@LaurenSpiegel: Such changes aren't the point of this PR.

**Comment by GuillaumeGomez** [comment]
cc @LaurenSpiegel

**Comment by LaurenSpiegel** [comment]
@GuillaumeGomez Has an end to end been run?

**Comment by GuillaumeGomez** [comment]
Ah my bad. Didn't see your message. What is an end to end? You mean calling ironman-machine?

**Comment by LaurenSpiegel** [comment]
@ironman-machine are you listening?

**Comment by LaurenSpiegel** [comment]
@ironman-machine this time with the dep patcher branches
SCALITY_FEDERATION_BRANCH=ft/dep-patcher-ansible
SCALITY_INTEGRATION_BRANCH=fix/nimrod-build-dep

**Comment by GuillaumeGomez** [comment]
@ironman-machine klsjfbgkjsfbgks
SCALITY_FEDERATION_BRANCH=ft/dep-patcher-ansible
SCALITY_INTEGRATION_BRANCH=fix/nimrod-build-dep

**Comment by GuillaumeGomez** [comment]
@ironman-machine 
SCALITY_FEDERATION_BRANCH=ft/dep-patcher-ansible
SCALITY_INTEGRATION_BRANCH=fix/nimrod-build-dep

**Comment by GuillaumeGomez** [comment]
@ironman-machine 
SCALITY_FEDERATION_BRANCH=ft/dep-patcher-ansible
SCALITY_INTEGRATION_BRANCH=fix/nimrod-build-dep

**Comment by GuillaumeGomez** [comment]
@ironman-machine 
SCALITY_FEDERATION_BRANCH=ft/dep-patcher-ansible
SCALITY_INTEGRATION_BRANCH=fix/nimrod-build-dep

**Comment by GuillaumeGomez** [comment]
Tests finally succeeded! \o/

**Comment by GuillaumeGomez** [comment]
cc @LaurenSpiegel

**Pull request merged**